pkg/apis/resource/v1alpha1/git/git_resource.go

```diff
@@ -147,6 +159,23 @@ func (s *Resource) GetInputTaskModifier(_ *v1alpha1.TaskSpec, path string) (v1al
 		args = append(args, "-sslVerify=false")
 	}
 
+	env := []corev1.EnvVar{{
+		Name:  "TEKTON_RESOURCE_NAME",
+		Value: s.Name,
+	}}
+
+	if len(s.HTTPProxy) != 0 {
+		env = append(env, corev1.EnvVar{Name: "HTTP_PROXY", Value: s.HTTPProxy})
```

Comment:
Please note that here, respective proxy env. variables are being set while creating a container instead of running `git config --global` like `sslVerify` in [git.go](https://github.com/tektoncd/pipeline/blob/master/pkg/git/git.go#L79)

The reason for implementing this way is, for some reason, `git` proxy configuration was ignored and cloning a repo was not failing even after setting `HTTPS_PROXY` to an invalid server, see the failure test at
https://github.com/tektoncd/pipeline/pull/2215/files#diff-6e725a312c3f914d42e49f045db0bde5R185